presto-spi/src/main/java/com/facebook/presto/spi/Constraint.java

```diff
@@ -50,7 +61,7 @@ public Constraint(TupleDomain<T> summary, Predicate<Map<T, NullableValue>> predi
         return summary;
     }
 
-    public Predicate<Map<T, NullableValue>> predicate()
+    public Optional<Predicate<Map<T, NullableValue>>> predicate()
```

Comment:
maybe extract some kind of `boolean testPredicate(Map<T, NullableValue> values)` method?